components/infobox/wikis/freefire/infobox_person_custom.lua

```diff
@@ -0,0 +1,177 @@
+	granader = {category = 'Granader', variable = 'Granader', isplayer = true},
+	igl = {category = 'In-game leaders', variable = 'In-game leader', isplayer = true},
+
+	--Staff and Talents
+	analyst = {category = 'Analysts', variable = 'Analyst', talent = true},
+	host = {category = 'Hosts', variable = 'Host', talent = true},
+	journalist = {category = 'Journalists', variable = 'Journalist', talent = true},
+	coach = {category = 'Coaches', variable = 'Coach', staff = true},
+	caster = {category = 'Casters', variable = 'Caster', talent = true},
+	manager = {category = 'Managers', variable = 'Manager', staff = true},
+	producer = {category = 'Producers', variable = 'Producer', talent = true},
+	streamer = {category = 'Streamers', variable = 'Streamer', talent = true},
+}
+
+local CustomPlayer = Class.new()
+local CustomInjector = Class.new(Injector)
+local _pagename = mw.title.getCurrentTitle().prefixedText
+local _args
+
+function CustomPlayer.run(frame)
+	local player = Player(frame)
+	_args = player.args
+
+	player.defineCustomPageVariables = CustomPlayer.defineCustomPageVariables
+	player.getPersonType = CustomPlayer.getPersonType
+	player.adjustLPDB = CustomPlayer.adjustLPDB
+	player.createWidgetInjector = CustomPlayer.createWidgetInjector
+
+	return player:createInfobox(frame)
+end
+
+function CustomInjector:parse(id, widgets)
+	if id == 'history' then
+		local manualHistory = _args.history
+		local automatedHistory = TeamHistoryAuto._results({
+			convertrole = 'true',
+			player = _pagename
+		}) or ''
+		automatedHistory = tostring(automatedHistory)
+		if automatedHistory == _EMPTY_AUTO_HISTORY then
+			automatedHistory = nil
+		end
+
+		if String.isNotEmpty(manualHistory) or String.isNotEmpty(automatedHistory) then
+			return {
+				Title{name = 'History'},
+				Center{content = {manualHistory}},
+				Center{content = {automatedHistory}},
+			        }
+		end
+	elseif id == 'role' then
+		return {
+			Cell{name = 'Role', content = {
+				CustomPlayer._createRole('role', _args.role),
+				CustomPlayer._createRole('role2', _args.role2)
+			}},
+		}
+	elseif id == 'status' then
+		return {
+			Cell{name = 'Status', content = CustomPlayer._getStatusContents()},
+			Cell{name = 'Years Active (Player)', content = {_args.years_active}},
+			Cell{name = 'Years Active (Org)', content = {_args.years_active_manage}},
+			Cell{name = 'Years Active (Coach)', content = {_args.years_active_coach}},
+		}
+	end
+	return widgets
+end
+
+function CustomPlayer:createWidgetInjector()
+	return CustomInjector()
+end
+
+function CustomPlayer:adjustLPDB(lpdbData)
+	lpdbData.extradata.isplayer = Variables.varDefault('isplayer')
+	lpdbData.extradata.role = Variables.varDefault('role')
+	lpdbData.extradata.role2 = Variables.varDefault('role2')
+
+	lpdbData.region = Template.safeExpand(mw.getCurrentFrame(), 'Player region', {_args.country})
+
+	if String.isNotEmpty(_args.team2) then
+		lpdbData.extradata.team2 = mw.ext.TeamTemplate.raw(_args.team2 or '')
```

Comment:
```suggestion
		lpdbData.extradata.team2 = (mw.ext.TeamTemplate.raw(_args.team2) or {}).page
```

possibly add a `or _args.team2` at the end too so in case the TT doesn't exist the raw string input gets stored

Reply:
wikicode used `_args.team2link or _args.team2` instead of flat `_args.team2`

Reply:
will it cause any errors to current wiki code used pages? if yes can we use the below code to match that?
`(_args.team2link) and _args.team2link or _args.team2 or '')`

Reply:
i would do something like this:
```lua
	local team2 = _args.team2link or _args.team2
 	if String.isNotEmpty(team2) then
 		lpdbData.extradata.team2 = (mw.ext.TeamTemplate.raw(team2) or {}).page or team2
 	end
```